src/components/button/button.tsx

```diff
@@ -0,0 +1,35 @@
+import cx from "classnames";
+import { HTMLAttributes } from "react";
+
+type Props = HTMLAttributes<HTMLButtonElement> & {
+  variant?: "primary" | "secondary" | "text" | "icon";
+  buttonText?: string;
+  icon?: string;
+  className?: string;
+};
+
+export const Button = ({
+  variant = "primary",
+  buttonText = "value",
+  icon,
+  className,
+  ...rest
+}: Props) => {
+  return (
+    <button 
```

Comment:
I really like the way this is handled and this certainly follows the pattern Josh was looking for.